src/core/lib/attribute-transition-manager.js

```diff
@@ -0,0 +1,353 @@
+import {GL, Program, Model, Geometry, Buffer, TransformFeedback} from 'luma.gl';
+
+const ATTRIBUTE_MAPPING = {
+  1: 'float',
+  2: 'vec2',
+  3: 'vec3',
+  4: 'vec4'
+};
+
+const TRANSITION_STATE = {
+  NONE: 0,
+  PENDING: 1,
+  STARTED: 2,
+  ENDED: 3
+};
+
+const noop = () => {};
+
+export default class AttributeTransitionManager {
+
+  static isSupported(gl) {
+    return TransformFeedback.isSupported(gl);
+  }
+
+  constructor({id, gl}, opts) {
+    this.id = id;
+    this.gl = gl;
+    this.opts = opts;
+
+    this.attributeTransitions = {};
+
+    this.needsRedraw = false;
+    this.model = null;
+    this.transformFeedback = new TransformFeedback(gl);
+  }
+
+  /* Public methods */
+  setOptions(opts) {
+    this.opts = opts || {};
+  }
+
+  // Called when attribute manager updates
+  // Extracts the list of attributes that need transition
+  update(attributes) {
```

Comment:
I liked the split into a separate transition manager class, however the two classes are tightly coupled here by both working on the same internal attribute array, which is a pretty complex object. We may want to address this, perhaps by making each object in that array into an internal object with methods.